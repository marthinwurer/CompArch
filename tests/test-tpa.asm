; test the tpa instruction
	.entry	start
;
	.org	0x20
zero:	.word	0
one:	.word	0x00001
five:	.word	0x00005
neg1:	.word	0xfffff
neg5:	.word	0xffffb
error:	.word	0x00bad
;
	.org	0x100
start:
	lda	zero
	tpa	err1
	tr	test2
err1:	lda	error
;
test2:	lda	one
	tpa	test3
err2:	lda	error
;
test3:	lda	five
	tpa	test4
err3:	lda	error
;
test4:	lda	neg1
	tpa	err4
	tr	test5
err4:	lda	error
;
test5:	lda	zero
	tpa	err5
	tr	test6
err5:	lda	error
;
test6:	lda	neg5
	tpa	err6
	tr	out
err6:	lda	error
;
out:
	halt
