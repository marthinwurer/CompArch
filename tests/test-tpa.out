CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 100
100:  10020  LDA   020  A[00000]  B[00000]  XR[000]
101:  36103  TPA   103  A[00000]  B[00000]  XR[000]
102:  30104  TR    104  A[00000]  B[00000]  XR[000]
104:  10021  LDA   021  A[00000]  B[00000]  XR[000]
105:  36107  TPA   107  A[00001]  B[00000]  XR[000]
107:  10022  LDA   022  A[00001]  B[00000]  XR[000]
108:  3610a  TPA   10a  A[00005]  B[00000]  XR[000]
10a:  10023  LDA   023  A[00005]  B[00000]  XR[000]
10b:  3610d  TPA   10d  A[fffff]  B[00000]  XR[000]
10c:  3010e  TR    10e  A[fffff]  B[00000]  XR[000]
10e:  10020  LDA   020  A[fffff]  B[00000]  XR[000]
10f:  36111  TPA   111  A[00000]  B[00000]  XR[000]
110:  30112  TR    112  A[00000]  B[00000]  XR[000]
112:  10024  LDA   024  A[00000]  B[00000]  XR[000]
113:  36115  TPA   115  A[ffffb]  B[00000]  XR[000]
114:  30116  TR    116  A[ffffb]  B[00000]  XR[000]
116:  00000  HALT       A[ffffb]  B[00000]  XR[000]

Machine Halted - HALT instruction executed

Simulated time 90 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
